From 1ca5844bb091ca729d9972d00c3ee16e37467ffe Mon Sep 17 00:00:00 2001
From: "David C. Manuelda" <StormByte@gmail.com>
Date: Thu, 3 Sep 2026 07:28:19 +0200
Subject: [PATCH 1/3] cmake: always emit libpng.pc, including MSVC

The UNIX/Cygwin guard skipped pkg-config on Windows. FFmpeg
Meson only sees this file. Keep the versioned and unversioned
names.
---
 CMakeLists.txt | 46 +++++++++++++++++++++-------------------------
 1 file changed, 21 insertions(+), 25 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69042960f..009074ebe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1088,21 +1088,19 @@ endif()
 # We use the same files like ./configure, so we have to set its vars.
 # Only do this on Windows for Cygwin - the files don't make much sense
 # outside of a UNIX look-alike.
-if(NOT WIN32 OR CYGWIN OR MINGW)
-  set(prefix "${CMAKE_INSTALL_PREFIX}")
-  set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
-  set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
-  set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
-  set(LIBS "-lz -lm")
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in"
-                 "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
-                 @ONLY)
-  create_symlink(libpng.pc FILE libpng${PNGLIB_ABI_VERSION}.pc)
-  configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng-config.in"
-                 "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
-                 @ONLY)
-  create_symlink(libpng-config FILE libpng${PNGLIB_ABI_VERSION}-config)
-endif()
+set(prefix "${CMAKE_INSTALL_PREFIX}")
+set(exec_prefix "${CMAKE_INSTALL_PREFIX}")
+set(libdir "${CMAKE_INSTALL_FULL_LIBDIR}")
+set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
+set(LIBS "-lz -lm")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng.pc.in"
+				"${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
+				@ONLY)
+create_symlink(libpng.pc FILE libpng${PNGLIB_ABI_VERSION}.pc)
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/libpng-config.in"
+				"${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
+				@ONLY)
+create_symlink(libpng-config FILE libpng${PNGLIB_ABI_VERSION}-config)
 
 # Install.
 if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
@@ -1158,16 +1156,14 @@ if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
   install(FILES png.5
           DESTINATION "${CMAKE_INSTALL_MANDIR}/man5")
   # Install the pkg-config files.
-  if(NOT CMAKE_HOST_WIN32 OR CYGWIN OR MINGW)
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng.pc"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
-            DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-    install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
-            DESTINATION "${CMAKE_INSTALL_BINDIR}")
-  endif()
+  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng.pc"
+          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng-config"
+          DESTINATION "${CMAKE_INSTALL_BINDIR}")
+  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}.pc"
+          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+  install(PROGRAMS "${CMAKE_CURRENT_BINARY_DIR}/libpng${PNGLIB_ABI_VERSION}-config"
+          DESTINATION "${CMAKE_INSTALL_BINDIR}")
 endif()
 
 # Create an export file that CMake users can include() to import our targets.
-- 
2.55.0

